flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -711,8 +701,18 @@ public FileVisitResult preVisitDirectory(java.nio.file.Path dir, BasicFileAttrib
 		LocalResource flinkConf = Records.newRecord(LocalResource.class);
 		Path remotePathJar =
 			Utils.setupLocalResource(fs, appId.toString(), flinkJarPath, appMasterJar, fs.getHomeDirectory());
-		Path remotePathConf =
-			Utils.setupLocalResource(fs, appId.toString(), flinkConfigurationPath, flinkConf, fs.getHomeDirectory());
+
+		final org.apache.flink.configuration.Configuration dynamicProperties = GlobalConfiguration.getDynamicProperties();
```

Comment:
Can't we pass this configuration into the `ClusterDescriptor` differently? What about reading it when creating the descriptor and then passing it as a constructor parameter? Even better, we could read the global configuration, then parse the dynamic properties, adding them to the read config and then creating the descriptor.